src/providers/navigation-state/navigation-helper.ts

```diff
@@ -0,0 +1,20 @@
+import { Injectable } from '@angular/core';
+import { App, NavController } from 'ionic-angular';
+
+/**
+ * Ionic will not allow the navigation controller to be injected at the root
+ * level, so this alternative approach gets the nav from the app.
+ *
+ * This helper allows the navigation controller to be mocked during testing.
+ */
+
+@Injectable()
+export class NavigationHelper {
```

Comment:
Can you add provider to the class name :)

Reply:
...and move it to a separate provider?

Reply:
Done. Makes more sense now.